### PR TITLE
 Updates league/commonmark to ^0.18

### PR DESCRIPTION
**Pull request by sanpii**

### CODE REVIEW

**Review by danhunsaker — APPROVED** [review]

**Comment by glensc** [comment]
@sanpii this seems to break compatibility

you should do like commonmark extras does:
- https://github.com/thephpleague/commonmark-extras/blob/8566350f490052fe81a70a7b4c18b43da222c7fc/composer.json#L18

```json
"^0.11|^0.12|^0.13|^0.14|^0.15|^0.16|^0.17|^0.18"
```

**Comment by glensc** [comment]
@sanpii also, you should never create PRs from master branch, it may add commits to PR you did not intend, as it's linked with your `master` branch.

**Comment by glensc** [comment]
@lossendae ping; @sanpii ping;

File: tests/RendererTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+use League\CommonMark\{
+    Converter,
+    DocParser,
+    Environment,
+    HtmlRenderer
+};
+use Lossendae\CommonMark\TaskLists\{
+    TaskListsParser,
+    TaskListsCheckboxRenderer
+};
```
Comment (glensc) [inline tests/RendererTest.php]: composer.json declares php 5.4, this definately is not supported there

**Comment by glensc** [comment]
looks to me this requires more changes

**Comment by glensc** [comment]
@lossendae you should probably indicate the direction:
- is it ok to support only 0.18?
- should php requirement bumped or the changes be compatible with php 5.4

**Comment by sanpii** [comment]
Thank you for your feedbacks. It’s fixed.

File: tests/RendererTest.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
+<?php declare(strict_types = 1);
```
Comment (glensc) [inline tests/RendererTest.php:1]: `declare(strict_types = 1);` is not available in php 5.4

File: composer.json [hunk 3]
```diff
@@ -14,13 +14,19 @@
     ],
     "require": {
         "php": ">=5.4",
-        "league/commonmark": "0.11.*"
+        "league/commonmark": ">=0.11,<0.19"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "^7.1"
```
Comment (glensc) [inline composer.json:20]: that version of phpunit is not php 5.4 compatible. altho for tests probably ok to have higher php requirement, even it seems super odd.

**Comment by glensc** [comment]
@colinodell here's another extension I rely on, but seems abandoned.

**Comment by glensc** [comment]
@colinodell did you notice my previous question?

my question would be: perhaps transfer this extension also to official group, then at least can add maintainers more easily even if original devs dissapear

for example this code renders funny:

```markdown
[x] ee
[x] lv
[x] lt
```

**Comment by glensc** [comment]
getting such error with 0.18 (can't upgrade due this extension not updated to newer)

```
Warning: Creating default object from empty value in 
vendor/league/commonmark/src/Node/Node.php on line 92
```

from such markup:
```markdown
*list*
[x] Ilmateade
```

and it's actually fatal error, as nothing is rendered, but not clear where it invoked die.

**Pull request merged**